core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -310,11 +330,52 @@ public void run() {
     }
   }
 
-  private ScheduledFuture<?> startDeadlineTimer(Deadline deadline) {
+  private ScheduledFuture<?> startDeadlineTimer(final Deadline deadline) {
```

Comment:
At the very least there's `FutureTask` and `Futures.immediateCancelledFuture()`.